stdlib/3/os/path.pyi

```diff
@@ -2,9 +2,15 @@
 # Ron Murawski <ron@horizonchess.com>
 
 # based on http://docs.python.org/3.2/library/os.path.html
-
+# adapted for 2.7 by Michal Pokorny
 import sys
-from typing import overload, List, Any, AnyStr, Sequence, Tuple, BinaryIO, TextIO
+from typing import (
+    overload, List, Any, AnyStr, Sequence, Tuple, BinaryIO, TextIO,
+    TypeVar, Union, Text, Callable
+)
+
+_T = TypeVar('_T')
+_PathType = Union[bytes, Text]
```

Comment:
Looked into this I think it need more work.  PathLike is 3.6 only and I imagine some methods support it.  But I'm not sure this is the change to work on that.